src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsBuildManager.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+
+using Microsoft.VisualStudio.ProjectSystem.VS.ConnectionPoint;
+using Microsoft.VisualStudio.ProjectSystem.VS.TempPE;
+using Microsoft.VisualStudio.ProjectSystem.VS.Utilities;
+
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Automation
+{
+    /// <summary>
+    /// Manages the portable executable (PE) files produced by running custom tools.
+    /// </summary>
+    [Export(typeof(BuildManager))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    [Order(Order.Default)]
+    internal class VSBuildManager : ConnectionPointContainer,
+                                    IEventSource<_dispBuildManagerEvents>,
+                                    BuildManager,
+                                    BuildManagerEvents
+    {
+        private readonly IProjectThreadingService _threadingService;
+        private readonly IUnconfiguredProjectCommonServices _unconfiguredProjectServices;
+        private readonly Lazy<ITempPEBuildManager> _tempPEManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VSBuildManager"/> class.
+        /// </summary>
+        [ImportingConstructor]
+        internal VSBuildManager(IProjectThreadingService threadingService, IUnconfiguredProjectCommonServices unconfiguredProjectServices,
+            Lazy<ITempPEBuildManager> tempPEManager)
+        {
+            AddEventSource(this);
+            _threadingService = threadingService;
+            _unconfiguredProjectServices = unconfiguredProjectServices;
+            _tempPEManager = tempPEManager;
+
+            Project = new OrderPrecedenceImportCollection<VSLangProj.VSProject>(projectCapabilityCheckProvider: _unconfiguredProjectServices.Project);
+        }
+
+        [ImportMany(ExportContractNames.VsTypes.VSProject)]
+        internal OrderPrecedenceImportCollection<VSLangProj.VSProject> Project { get; }
+
+        /// <summary>
+        /// Occurs when a design time output moniker is deleted.
+        /// </summary>
+        public event _dispBuildManagerEvents_DesignTimeOutputDeletedEventHandler DesignTimeOutputDeleted;
+
+        /// <summary>
```

Comment:
We don't program against this type, so these comments are already represented by the interface it implements, and hence we don't need them.